pandas/tests/test_downstream.py

```diff
@@ -283,6 +295,11 @@ def test_missing_required_dependency():
     # We skip this test if pandas is installed as a site package. We first
     # import the package normally and check the path to the module before
     # executing the test which imports pandas with site packages disabled.
+
+    # Can't import pandas from the test directory since its not
+    # built inplace with meson
+    if pd._built_with_meson:
```

Comment:
Is this expected to be part of the final design or just a hack for now? I think we should just commit to meson as the only build option and avoid branching like this

Reply:
Just a hack, will be removed when setuptools is removed.

The monkeypatching will be here to stay though, because meson does out-of-tree builds.